### PR TITLE
Code Smell Merge

### PR DESCRIPTION
**Pull request by adamyafout**

### CODE REVIEW

**Review by car12lin12 — COMMENTED** [review]
Great job Adam!! Very nice method extractions. I'm not sure we can remove the todo yet so I'll let Zubair approve that

File: app/mapUrls.go [hunk 1]
```diff
@@ -26,7 +27,8 @@ func mapUrls(mw Middleware, r *gin.Engine, mh *http.MessageHandler, rh *http.Roo
 	router.PUT("/rooms/remove/:roomID/:id", rh.RemoveUserFromRoom)
 	router.DELETE("/rooms/:roomID", rh.DeleteRoom)
 
-	router.GET("chat/:roomID", mh.LoadMessages)
-	router.PUT("chat/:roomID", mh.EditMessage)
-	router.DELETE("chat/:roomID", mh.DeleteMessage)
+
+	router.GET(pathRoomID, mh.LoadMessages)
```
Comment (nuzurie) [inline app/mapUrls.go:29]: move line 10 to line 30 please for better readability

File: messaging/delivery/http/chat_delivery.go [hunk 2]
```diff
@@ -191,23 +194,9 @@ func (h *hub) StartHubListener() {
...
-			if connections != nil {
-				if _, ok := connections[s]; ok {
-					delete(connections, s)
-					close(s.conn.send)
-					if len(connections) == 0 {
-						delete(h.rooms, s.roomID)
-					}
-				}
-			}
+			h.UnregisterCase(s)
```
Comment (nuzurie) [inline messaging/delivery/http/chat_delivery.go:199]: good job!

File: messaging/delivery/http/chat_delivery.go [hunk 3]
```diff
@@ -229,6 +218,28 @@ func (h *hub) StartHubListener() {
...
+func (h *hub) RegisterCase(s subscription) {
+	connections := h.rooms[s.roomID]
+	if connections == nil {
+		connections = make(map[subscription]bool)
+		h.rooms[s.roomID] = connections
+	}
+	h.rooms[s.roomID][s] = true
+}
+
+func (h *hub) UnregisterCase(s subscription) {
```
Comment (nuzurie) [inline messaging/delivery/http/chat_delivery.go:230]: Really impressive extraction!

**Review by nuzurie — APPROVED** [review]

**Comment by nuzurie** [comment]
Approved. Please make one small change, and we're good :)

**Comment by adamyafout** [comment]
The request changes made @nuzurie

**Pull request merged**